### DISCUSSION TITLE
Welcome to desky Discussions!

### DISCUSSION THREAD

**Discussion by developit** [developit/desky#1] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by developit** [comment]
cc @vipial1 @stvnwn @jgissend10 @vkhurana

**Reply by vipial1** [reply]
Regarding supporting your microcontroller and the buttons at the same time...you suggested to simply solder both connections, isn't it?
Are you sure that simply soldering the cables together will work? that protocol is not designed for bus. Also, I don't know how the buttons will behave when receive commands that doesn't expect.

**Reply by jtszalay** [reply]
Yeah this is the thing I'm still trying to figure out. How to support both the microcontroller and the buttons at the same time. I've got a stream deck on my desk that I've been using to make API calls in the mean time.

**Reply by piitaya** [reply]
Hi 👋
I found this repository : https://github.com/adfinis-sygroup/muuvctl#using-controller-while-usb-plugged-in. 
Looks like a 5V relay can be used to used the buttons and the microcontroller at the same time but I didn't test it and my knowledge is very limited 😅

**Comment by PaulAlarcon** [comment]
So excited to hear more about automating the autonomous desk - I'm a developer who thinks this could help in many ergonomic and fitness ways. I could see the infrastructure of the software involved but can't seem to understand the physical components - I'm really trying to read more about it and get into it - any guides you guys recommend to understand the ESP microchips ? Maybe a side project to get started with?

**Comment by BoTime** [comment]
Thank you @developit, with the information you shared, I could control my Autonomous desk from Alexa. Considering the network delay, I only allow change to preset height, similar to pressing button 1 and button 2.

The hardware I am using is Arduino Nano 33 IoT. Need to set up IFTTT, Arduino IoT Cloud, and AWS Lambda. If anyone is interested, I can share more details.

**Comment by vipial1** [comment]
@developit @BoTime 
I finally found some time to work in integrating this with Home Assistant, but I am not able to build desky project, it looks like a util file is missing.
This file https://gist.github.com/developit/d610e45a522810b5287db61e554ae9c9#file-server-js is importing some buffer object:
`import { buffer } from './util.js';`

but that util.js is not part of the project.

Can you please @developit upload it?

Thanks.

**Reply by developit** [reply]
Ah! Sorry about that, I'll try to get it added today. Not sure what happened.

**Reply by developit** [reply]
@vipial1 Alright, I added that file to the Gist:
https://gist.github.com/developit/d610e45a522810b5287db61e554ae9c9#file-util-js

Also, FWIW I just decided to Open Source [espz](https://github.com/developit/espz), and the repo includes copies of a few simple Microcontroller projects, including Desky:

https://github.com/developit/espz/tree/main/demo/desk

**Comment by vipial1** [comment]
And finally something working...if you want to try it guys:

https://github.com/vipial1/desky-ha

**Reply by developit** [reply]
ooh nice! @vipial1 where did you place the diode? on `tx`? It's been a painfully long time since I did any real electronics work haha.

**Comment by lizaoreo** [comment]
Trying to figure out if this will work with my TiMotion desk, what controller (the black box under the desk the motors connect to) are you guys interfacing with?  I've not been able to get espruino working quite right to test yet, but my desk only came with a hand unit with up/down buttons, so I'm not even sure this would work for adding memory if I do get it working.  My desks controller is a TC15.

**Reply by developit** [reply]
@jtszalay the controller for Autonomous desks is TiMotion model number TC11-Z72. As best I can tell from browsing around the TiMotion website, all of their controllers use the same protocol - they even sell their own app/bluetooth adapter [here](https://www.timotion.com/en/products/accessories/twd1-series#features). That should mean your TC15 would work with this software.

**Reply by lizaoreo** [reply]
Yeah, I feel like it should work, though another guy I was chatting with used ESPHome to get his TC15 to work.  Much like me, he has a up/down only remote and I assumed he was using the UART stuff for ESPHome, but he's actually just sending signals to simulate pressing the buttons to the controller.  I can get Espruino flashed to the ESP32 mini I'm using to fiddle with (failed on the 8266 I tried), but after I push desky to it, I can never get into a web interface or anything else.  I'll have to mess around some more with that part.

Thanks for responding.
